src/main/java/appeng/util/inv/AdaptorItemHandler.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * This file is part of Applied Energistics 2.
+ * Copyright (c) 2013 - 2014, AlgorithmX2, All rights reserved.
+ *
+ * Applied Energistics 2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.util.inv;
+
+
+import java.util.Iterator;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.config.FuzzyMode;
+import appeng.util.InventoryAdaptor;
+import appeng.util.Platform;
+
+
+public class AdaptorItemHandler extends InventoryAdaptor
+{
+	private final IItemHandler itemHandler;
+
+	public AdaptorItemHandler( IItemHandler itemHandler )
+	{
+		this.itemHandler = itemHandler;
+	}
+
+	@Override
+	public ItemStack removeItems( int amount, ItemStack filter, IInventoryDestination destination )
+	{
+		int slots = itemHandler.getSlots();
+		ItemStack rv = null;
+
+		for( int slot = 0; slot < slots && amount > 0; slot++ )
+		{
+			final ItemStack is = itemHandler.getStackInSlot( slot );
+			if( is == null || ( filter != null && !Platform.isSameItemPrecise( is, filter ) ) )
+			{
+				continue;
+			}
+
+			if( destination != null )
+			{
+				ItemStack extracted = itemHandler.extractItem( slot, amount, true );
+				if( extracted == null )
+				{
+					continue;
+				}
+
+				if( !destination.canInsert( extracted ) )
+				{
+					continue;
+				}
+			}
+
+			// Attempt extracting it
+			ItemStack extracted = itemHandler.extractItem( slot, amount, false );
+
+			if( extracted == null )
+			{
+				continue;
+			}
+
+			if( rv == null )
+			{
+				// Use the first stack as a template for the result
+				rv = extracted;
+				filter = extracted;
+				amount -= extracted.stackSize;
+			}
+			else
+			{
+				// Subsequent stacks will just increase the extracted size
+				rv.stackSize += extracted.stackSize;
+				amount -= extracted.stackSize;
+			}
+		}
+
+		return rv;
+	}
+
+	@Override
+	public ItemStack simulateRemove( int amount, ItemStack filter, IInventoryDestination destination )
+	{
+		int slots = itemHandler.getSlots();
+		ItemStack rv = null;
+
+		for( int slot = 0; slot < slots && amount > 0; slot++ )
+		{
+			final ItemStack is = itemHandler.getStackInSlot( slot );
+			if( is != null && ( filter == null || Platform.isSameItemPrecise( is, filter ) ) )
+			{
+				ItemStack extracted = itemHandler.extractItem( slot, amount, true );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'extracted' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)